python_modules/libraries/dagster-azure/dagster_azure/adls2/fake_adls2_resource.py

```diff
@@ -83,9 +114,13 @@ def has_file(self, path):
         return bool(self._file_system.get(path))
 
     def get_file_client(self, file_path):
+        # pass fileclient a ref to self and its name so the file can delete itself
+        self._file_system.setdefault(file_path, FakeADLS2FileClient(self, file_path))
```

Comment:
Can you explain why it's necessary to pass FileClient a reference to self / which file is deleting itself? I'm having trouble following.

Reply:
I see. Having the backref so that the file system client can manage its own state makes sense.